pkg/util/file_indexer.go

```diff
@@ -8,35 +8,59 @@ import (
 	"time"
 
 	"github.com/golang/glog"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 const fileIndexDirectory = ".odo"
 const fileIndexName = "odo-file-index.json"
 
+// FileIndex holds the file index used for storing local file state change
+type FileIndex struct {
+	metav1.TypeMeta
+	FileMap map[string]FileData
+}
+
+// NewFileIndex returns a fileIndex
+func NewFileIndex() *FileIndex {
+
+	return &FileIndex{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "FileIndex",
```

Comment:
OdoFileIndex

Reply:
The reason for not adding `Odo` as because the `Kind` for preference is `Preference` and not `OdoPreference` hence wanted to keep things uniform